packages/ethereum-contracts/contracts/identity/IdentityRegistry.sol

```diff
@@ -42,9 +42,9 @@ contract IdentityRegistry is Ownable {
             ethIdentityAccounts != address(0),
             "Ethereum identity accounts is not set"
         );
-        require(identities[msg.sender] == address(0), "Identity exists");
         Identity newIdentity = new Identity(msg.sender, keyEnums);
-        identities[msg.sender] = newIdentity;
+        require(identities[newIdentity] == false, "Identity exists");
+        identities[newIdentity] = true;
 
         ethIdentityAccounts.newIdentity(newIdentity, msg.sender);
```

Comment:
newIdentity(address identity, address sender), for new identity, the identity no need to be the same as msg.sender?
I see newIdentity emits AccountAdded(identity, account) event, so how is this different from addAccount